src/app/graphicsUtils.js

```diff
@@ -0,0 +1,31 @@
+define([
+  'esri/graphicsUtils'
+], function (
+  graphicsUtils
+) {
+    return {
+        unionGraphicsIntoExtent: function (graphics) {
```

Comment:
This function can be replaced entirely by graphicsUtils.graphicsExtent which takes an array of graphics as an input, can't it? Or are you doing some extra error checking?